streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamPartitionAssignor.java

```diff
@@ -593,9 +592,13 @@ public void onAssignment(Assignment assignment) {
     private void prepareTopic(final Map<String, InternalTopicMetadata> topicPartitions) {
         log.debug("stream-thread [{}] Starting to validate internal topics in partition assignor.", streamThread.getName());
 
-        for (final Map.Entry<String, InternalTopicMetadata> entry : topicPartitions.entrySet()) {
-            final InternalTopicConfig topic = entry.getValue().config;
-            final Integer numPartitions = entry.getValue().numPartitions;
+        // first construct the topics to make ready
+        Map<InternalTopicConfig, Integer> topicsToMakeReady = new HashMap<>();
+        Set<String> topicNamesToMakeReady = new HashSet<>();
```

Comment:
Feels like this collection is redundant. You can get the name from `InternalTopicConfig` perhaps?